lib/board.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class Board
+  def initialize(layout:)
+    @layout = layout
+  end
+
+  def is_full(tokens:)
+    cells_equal_to_a_token = ->(cell) { tokens.include?(cell) }
+    @layout.all?(&cells_equal_to_a_token)
+  end
+
+  def at(position:)
+    @layout[position - 1]
+  end
+
+  def spaces_at(row:)
+    number_of_columns_in_row = Math.sqrt(@layout.length)
```

Comment:
😆 I think I end up going on this mini-tirade on every single apprentice repository but...

Calculating the number of rows/columns of the board using square root, plus allowing the board to be initialized with an array of any integer length, leaves you open to the error where e.g. someone could pass an array of length 12 to create a board and get a column number that is an irrational number.

By contrast, if you pass in the number of rows, retain that as an attribute of the board, and square it to get the full number of elements in your array, you not only avoid this error, but you also explicitly demonstrate that one of the properties of a tic-tac-toe board is that it is square.

Reply:
I don't quite understand what this method is for. Could you clarify?

Reply:
@mrfishball It returns the tokens in a given row. 

@hnlee @mrfishball I decided to remove this function because I don't use it anywhere. I think I was planning on using it to output the board, but a dynamic way to display the board ins't needed atm.